nesta/packages/companies_house/collect_ch_data_dump.py

```diff
@@ -0,0 +1,87 @@
+"""
+Finds the latest data-dump URL from Companies House, downloads it, and processes it.
+
+TODO: Data dumps are available for previous months in a year too?
+TODO: Deal with other columns
+TODO: Merge with NSPL
+"""
+import datetime
+import logging
+import os
+
+import engarde.checks as edc
+import numpy as np
+import pandas as pd
+import requests
+
+
+def _data_dump_url(date: datetime) -> str:
```

Comment:
I'm not against function annotation, but unless you feel strongly in favour of them (which you might do) I'd prefer to avoid mixing styles. Feel free to ignore!